src/services/cooperation.js

```diff
@@ -21,36 +22,14 @@ const cooperationService = {
     return result
   },
 
-  getCooperationById: async (id) => {
-    return await (
-      await Cooperation.findById(id)
-    ).populate([
-      { path: 'sections.resources.resource', select: '-createdAt -updatedAt' },
-      {
-        path: 'offer',
-        populate: [
-          {
-            path: 'category',
-            select: ['name', 'appearance']
-          },
-          {
-            path: 'subject',
-            select: 'name'
-          },
-          {
-            path: 'author',
-            select: ['firstName', 'lastName', 'photo', 'professionalSummary', 'totalReviews', 'FAQ', 'averageRating']
-          }
-        ],
-        select: ['id', 'author', 'category', 'subject', 'title', 'languages', 'proficiencyLevel', 'description']
-      },
-      {
-        path: 'initiator'
-      },
-      {
-        path: 'receiver'
-      }
-    ])
+  getCooperationById: async (id, userRole) => {
+    const isClosedResourcesHidden = userRole === roles.STUDENT
+
+    const pipeline = getCooperationByIdPipeline(id, isClosedResourcesHidden)
```

Comment:
why did you added the `pipeline` to naming?

Reply:
This function returns the pipeline for the aggregation query.
Do you think, that the name `getCooperationByIdQueryPipeline` would be a better name for this function?

Reply:
the name that includes `query` is more understandable, in my opinion, so yes, you can change it